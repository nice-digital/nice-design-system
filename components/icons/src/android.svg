<?xml version="1.0" encoding="utf-8"?>
<svg width="512" height="512" viewBox="0 0 512 512" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M199.704 127.577C202.575 127.577 205.042 126.545 207.106 124.481C209.17 122.417 210.202 119.949 210.202 117.077C210.202 114.205 209.17 111.737 207.106 109.673C205.042 107.609 202.575 106.577 199.704 106.577C196.833 106.577 194.41 107.609 192.436 109.673C190.462 111.737 189.475 114.205 189.475 117.077C189.475 119.949 190.462 122.417 192.436 124.481C194.41 126.545 196.833 127.577 199.704 127.577ZM313.296 127.577C316.167 127.577 318.59 126.545 320.564 124.481C322.538 122.417 323.525 119.949 323.525 117.077C323.525 114.205 322.538 111.737 320.564 109.673C318.59 107.609 316.167 106.577 313.296 106.577C310.425 106.577 307.957 107.609 305.894 109.673C303.83 111.737 302.798 114.205 302.798 117.077C302.798 119.949 303.83 122.417 305.894 124.481C307.957 126.545 310.425 127.577 313.296 127.577ZM94.7251 177.115C102.262 177.115 108.722 179.808 114.106 185.192C119.489 190.577 122.181 197.038 122.181 204.577V320.346C122.181 328.064 119.534 334.615 114.24 340C108.947 345.385 102.442 348.077 94.7251 348.077C87.0088 348.077 80.4588 345.385 75.0753 340C69.6918 334.615 67 328.064 67 320.346V204.577C67 197.038 69.6918 190.577 75.0753 185.192C80.4588 179.808 87.0088 177.115 94.7251 177.115ZM380.052 182.231V361.538C380.052 369.795 377.181 376.795 371.438 382.538C365.696 388.282 358.787 391.154 350.712 391.154H330.523V452.269C330.523 459.987 327.832 466.538 322.448 471.923C317.065 477.308 310.515 480 302.798 480C295.082 480 288.532 477.308 283.148 471.923C277.765 466.538 275.073 459.987 275.073 452.269V391.154H237.927V452.269C237.927 459.987 235.235 466.538 229.852 471.923C224.468 477.308 217.918 480 210.202 480C202.665 480 196.205 477.308 190.821 471.923C185.437 466.538 182.746 459.987 182.746 452.269L182.477 391.154H162.558C154.303 391.154 147.304 388.282 141.562 382.538C135.819 376.795 132.948 369.795 132.948 361.538V182.231H380.052ZM317.603 73.1923C336.804 83.0641 352.147 96.8397 363.632 114.519C375.117 132.199 380.859 151.538 380.859 172.538H131.871C131.871 151.538 137.614 132.199 149.099 114.519C160.584 96.8397 176.016 83.0641 195.397 73.1923L176.286 37.9231C175.029 35.5897 175.478 33.7949 177.631 32.5385C179.964 31.4615 181.759 32 183.015 34.1538L202.396 69.6923C219.443 62.1538 237.478 58.3846 256.5 58.3846C275.522 58.3846 293.557 62.1538 310.604 69.6923L329.985 34.1538C331.241 32 333.036 31.4615 335.369 32.5385C337.522 33.7949 337.971 35.5897 336.714 37.9231L317.603 73.1923ZM446 204.577V320.346C446 328.064 443.308 334.615 437.925 340C432.541 345.385 425.991 348.077 418.275 348.077C410.738 348.077 404.278 345.385 398.894 340C393.511 334.615 390.819 328.064 390.819 320.346V204.577C390.819 196.859 393.511 190.353 398.894 185.058C404.278 179.763 410.738 177.115 418.275 177.115C425.991 177.115 432.541 179.763 437.925 185.058C443.308 190.353 446 196.859 446 204.577Z" fill="black"/>
</svg>
