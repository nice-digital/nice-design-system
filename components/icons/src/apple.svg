<?xml version="1.0" encoding="utf-8"?>
<svg width="512" height="512" viewBox="0 0 512 512" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M441 358.004C434.113 380.09 423.252 402.175 408.418 424.26C385.637 458.89 362.944 476.205 340.34 476.205C331.686 476.205 319.324 473.378 303.254 467.724C288.067 462.07 274.734 459.243 263.255 459.243C252.482 459.243 239.944 462.159 225.64 467.989C211.335 473.996 199.68 477 190.673 477C163.831 477 137.253 454.12 110.94 408.359C84.9799 362.245 72 317.809 72 275.052C72 234.768 81.9777 201.728 101.933 175.933C121.889 150.49 146.966 137.769 177.164 137.769C189.879 137.769 205.508 140.419 224.05 145.72C242.416 151.02 254.602 153.671 260.606 153.671C268.553 153.671 281.179 150.667 298.486 144.66C316.499 138.653 331.775 135.649 344.313 135.649C365.328 135.649 384.136 141.391 400.736 152.876C409.919 159.236 419.102 168.07 428.285 179.378C414.334 191.216 404.268 201.64 398.087 210.651C386.608 227.259 380.869 245.546 380.869 265.511C380.869 287.419 386.961 307.12 399.146 324.611C411.332 342.103 425.283 353.234 441 358.004ZM341.399 47.131C341.399 57.9087 338.838 69.9231 333.717 83.1743C328.419 96.4255 320.207 108.617 309.082 119.748C299.546 129.288 290.009 135.649 280.473 138.829C273.939 140.773 264.756 142.275 252.924 143.335C253.454 117.009 260.341 94.3053 273.586 75.2236C286.654 56.3185 308.729 43.244 339.81 36C339.986 36.53 340.207 37.5018 340.472 38.9153C340.737 40.3287 340.958 41.3005 341.134 41.8305C341.134 42.5373 341.178 43.4207 341.267 44.4808C341.355 45.5409 341.399 46.4243 341.399 47.131Z" fill="black"/>
</svg>
