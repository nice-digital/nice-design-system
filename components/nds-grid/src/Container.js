import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import "../scss/container.scss";

export const Container = props => {
	const {
		children,
		className,
		elementType: ElementType = "div",
		fullWidth,
		...rest
	} = props;

	const classes = classnames([
		"container",
		fullWidth && "container--full",
		className
	]);

	return (
		<ElementType className={classes} {...rest}>
			{children}
		</ElementType>
	);
};

Container.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.arrayOf(PropTypes.node),
		PropTypes.node
	]).isRequired,
	className: PropTypes.string,
	elementType: PropTypes.elementType,
	fullWidth: PropTypes.bool
};
