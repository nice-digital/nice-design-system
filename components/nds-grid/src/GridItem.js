import React from "react";
import PropTypes from "prop-types";

export const GridItem = props => {
	const {
		children,
		cols,
		push,
		pull,
		xs,
		sm,
		md,
		lg,
		xl,
		className,
		elementType: GridItemElementType,
		...rest
	} = props;

	let gridColsAttr = [cols.toString()];

	if (push) gridColsAttr.push("push:" + push);
	if (pull) gridColsAttr.push("pull:" + pull);

	const bpGridDefs = { xs, sm, md, lg, xl };
	const mapBpToAttr = bp => {
		const bpGridDef = bpGridDefs[bp];

		if (!bpGridDef) return null;

		// E.g. "xs:6"
		if (typeof bpGridDef === "number") return bp + ":" + bpGridDef;

		// E.g. "xs:6 xs:push:3"
		const { cols, pull, push } = bpGridDef;
		let gridDefAttr = bp + ":" + cols;
		if (pull) gridDefAttr += ` ${bp}:pull:${pull}`;
		if (push) gridDefAttr += ` ${bp}:push:${push}`;
		return gridDefAttr;
	};
	gridColsAttr = gridColsAttr
		.concat(Object.keys(bpGridDefs).map(mapBpToAttr))
		.filter(col => col)
		.join(" ");

	return (
		<GridItemElementType data-g={gridColsAttr} className={className} {...rest}>
			{children}
		</GridItemElementType>
	);
};

const breakpointPropType = PropTypes.oneOfType([
	PropTypes.number,
	PropTypes.shape({
		cols: PropTypes.number.isRequired,
		push: PropTypes.number,
		pull: PropTypes.number
	})
]);

GridItem.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.arrayOf(PropTypes.node),
		PropTypes.node
	]).isRequired,
	cols: PropTypes.number.isRequired,
	push: PropTypes.number,
	pull: PropTypes.number,
	xs: breakpointPropType,
	sm: breakpointPropType,
	md: breakpointPropType,
	lg: breakpointPropType,
	xl: breakpointPropType,
	className: PropTypes.string,
	elementType: PropTypes.elementType
};

GridItem.defaultProps = {
	elementType: "div"
};
