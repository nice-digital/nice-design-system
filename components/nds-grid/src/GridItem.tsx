import React from "react";

export type Columns = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
export type PullOrPush = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11;
export type BreakPointGridDefinition =
	| Columns
	| {
			/** The number of columns at this breakpoint */
			cols: Columns;
			/** The number of columns to pull at this breakpoint */
			pull?: PullOrPush;
			/** The number of columns to push at this breakpoint */
			push?: PullOrPush;
	  };

export interface GridItemProps {
	/** Contents of the grid item */
	children: React.ReactNode;
	/** The number of columns for the first (smallest) screen size */
	cols?: Columns;
	/** The number of columns to pull the grid item for the first (smallest) screen size */
	pull?: PullOrPush;
	/** The number of columns to push the grid item for the first (smallest) screen size */
	push?: PullOrPush;
	/** Grid definition from the xs breakpoint */
	xs?: BreakPointGridDefinition;
	/** Grid definition from the sm breakpoint */
	sm?: BreakPointGridDefinition;
	/** Grid definition from the md breakpoint */
	md?: BreakPointGridDefinition;
	/** Grid definition from the lg breakpoint */
	lg?: BreakPointGridDefinition;
	/** Grid definition from the xl breakpoint */
	xl?: BreakPointGridDefinition;
	/** Additional classes to add to the grid, for example mt--e */
	className?: string;
	/** The type of DOM node to render for the grid item. Leave blank to default to div. */
	elementType?: React.ElementType;
}

export const GridItem: React.FC<GridItemProps> = (props) => {
	const {
		children,
		cols = 12,
		push,
		pull,
		xs,
		sm,
		md,
		lg,
		xl,
		className,
		elementType: GridItemElementType = "div",
		...rest
	} = props;

	let gridColsAttr: string | string[] = [cols.toString()];

	if (push) gridColsAttr.push("push:" + push);
	if (pull) gridColsAttr.push("pull:" + pull);

	const bpGridDefs = { xs, sm, md, lg, xl };
	const mapBpToAttr = (bp: string): string | null => {
		const key = bp as keyof typeof bpGridDefs;
		const bpGridDef: BreakPointGridDefinition | undefined = bpGridDefs[key];

		if (!bpGridDef) return null;

		// E.g. "xs:6"
		if (typeof bpGridDef === "number") return bp + ":" + bpGridDef;

		// E.g. "xs:6 xs:push:3"
		const { cols, pull, push } = bpGridDef;
		let gridDefAttr = bp + ":" + cols;
		if (pull) gridDefAttr += ` ${bp}:pull:${pull}`;
		if (push) gridDefAttr += ` ${bp}:push:${push}`;
		return gridDefAttr;
	};

	const bpGridDefKeys = Object.keys(bpGridDefs).map(mapBpToAttr) as string[];

	gridColsAttr = gridColsAttr
		.concat(bpGridDefKeys)
		.filter((col) => col)
		.join(" ");

	return (
		<GridItemElementType
			data-g={gridColsAttr}
			data-component="grid-item"
			className={className}
			{...rest}
		>
			{children}
		</GridItemElementType>
	);
};
