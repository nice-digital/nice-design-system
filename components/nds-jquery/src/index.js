import pluginAutoLoader from "./plugin-autoloader";
import Tabs from "./tabs.js";
import InPageNav from "./in-page-nav.js";
import Tracker from "./tracker";
import pluginizr from "./pluginizr";
import eventr from "./eventr";
import breakpoints from "../../nds-core/src/breakpoints";
import utils from "../../nds-core/src/utils";

let nice = {
	init: (el: $) => {

		// Load all component modules from this directory automatically, see http://stackoverflow.com/a/31770875/486434
		// But exclude test files http://stackoverflow.com/a/30372240
		pluginAutoLoader.load(require.context("./../components/", true, /^((?!test\.).)*\.js$/im));
		pluginAutoLoader.load(require.context("./", true, /^((?!test\.).)*\.js$/im));

		pluginAutoLoader.findPlugins(el);
	},
	// Components
	InPageNav: InPageNav,
	Tabs: Tabs,
	Tracker: Tracker,
	// Utils
	pluginAutoLoader: pluginAutoLoader,
	pluginizr: pluginizr,
	eventr: eventr,
	breakpoints,
	utils
};

$.fn.nice = function() {
	nice.init(this);
};

// Export to global namespace for precompiled usage
window.NICE = window.NICE || {};
window.NICE.nice = nice;

export {
	nice as default,
	// Components
	InPageNav,
	Tabs,
	Tracker,
	// Utils
	pluginAutoLoader,
	pluginizr,
	eventr,
	breakpoints,
	utils
};
