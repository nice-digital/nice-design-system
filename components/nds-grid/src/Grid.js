import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import { Container } from "./Container";
import { GridItem } from "./GridItem";
import "../scss/grid.scss";
export { GridItem };
export { Container };

export const Grid = props => {
	const {
		children,
		reverse,
		equalHeight,
		gutter,
		horizontalAlignment,
		verticalAlignment,
		debug,
		className,
		elementType: GridElementType,
		...rest
	} = props;

	const classNames = classnames({
		grid: true,
		"grid--rev": reverse,
		"grid--equal-height": equalHeight,
		"grid--gutterless": gutter === Grid.gutter.none,
		"grid--compact": gutter === Grid.gutter.compact,
		"grid--loose": gutter === Grid.gutter.loose,
		"grid--center": horizontalAlignment === Grid.horizontalAlignment.center,
		"grid--right": horizontalAlignment === Grid.horizontalAlignment.right,
		"grid--middle": verticalAlignment === Grid.verticalAlignment.middle,
		"grid--bottom": verticalAlignment === Grid.verticalAlignment.bottom,
		"grid--debug": debug,
		[className]: className
	});
	return (
		<GridElementType className={classNames} {...rest}>
			{children}
		</GridElementType>
	);
};

Grid.gutter = {
	none: "none",
	standard: "standard",
	compact: "compact",
	loose: "loose"
};

Grid.horizontalAlignment = {
	left: "left",
	center: "center",
	right: "right"
};

Grid.verticalAlignment = {
	top: "top",
	middle: "middle",
	bottom: "bottom"
};

Grid.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.arrayOf(PropTypes.element),
		PropTypes.element
	]).isRequired,
	reverse: PropTypes.bool,
	equalHeight: PropTypes.bool,
	horizontalAlignment: PropTypes.oneOf([
		Grid.horizontalAlignment.left,
		Grid.horizontalAlignment.center,
		Grid.horizontalAlignment.right
	]),
	verticalAlignment: PropTypes.oneOf([
		Grid.verticalAlignment.top,
		Grid.verticalAlignment.middle,
		Grid.verticalAlignment.bottom
	]),
	gutter: PropTypes.oneOf([
		Grid.gutter.standard,
		Grid.gutter.none,
		Grid.gutter.compact,
		Grid.gutter.loose
	]),
	debug: PropTypes.bool,
	className: PropTypes.string,
	elementType: PropTypes.elementType
};

Grid.defaultProps = {
	reverse: false,
	equalHeight: false,
	gutter: Grid.gutter.standard,
	horizontalAlignment: Grid.horizontalAlignment.left,
	verticalAlignment: Grid.verticalAlignment.top,
	elementType: "div"
};
