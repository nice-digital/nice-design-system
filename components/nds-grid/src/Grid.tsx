import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import { GridItem } from "./GridItem";
import "../scss/grid.scss";
export { GridItem };

type ValidGridItemProps =
	| React.ReactElement<typeof GridItem>
	| boolean
	| null
	| undefined;

export const gutterTypes = {
	standard: "standard",
	none: "none",
	compact: "compact",
	loose: "loose"
} as const;

export const horizontalAlignmentTypes = {
	left: "left",
	center: "center",
	right: "right"
} as const;

export const verticalAlignmentTypes = {
	top: "top",
	middle: "middle",
	bottom: "bottom"
} as const;

export interface GridProps {
	/** Grid item elements */
	children: ValidGridItemProps[] | ValidGridItemProps;
	/** Renders grid items in the opposite way to the source order */
	reverse?: boolean;
	/** Make children of grid items ahve 100% height to fill the vertical space */
	equalHeight?: boolean;
	/** The horizontal alignment of items within the grid, when there are empty columns. Leave blank to default to left. */
	horizontalAlignment?: keyof typeof horizontalAlignmentTypes;
	/** The vertical alignment of items within the grid. Leave blank to default to top. */
	verticalAlignment?: keyof typeof verticalAlignmentTypes;
	/** The gap between grid cells. Leave blank to default to the standard gutter. */
	gutter?: keyof typeof gutterTypes;
	/** Debug puts a coloured outline around the grid and its cells */
	debug?: boolean;
	/** Additional classes to add to the grid, for example mt--e */
	className?: string;
	/** The type of DOM node to render for the grid. Leave blank to default to div. */
	elementType?: React.ElementType;
}

export const Grid = (props: GridProps) => {
	const {
		children,
		reverse,
		equalHeight,
		gutter,
		horizontalAlignment,
		verticalAlignment,
		debug,
		className,
		elementType,
		...rest
	} = props;

	const classNames = classnames({
		grid: true,
		"grid--rev": reverse,
		"grid--equal-height": equalHeight,
		"grid--gutterless": gutter === gutterTypes.none,
		"grid--compact": gutter === gutterTypes.compact,
		"grid--loose": gutter === gutterTypes.loose,
		"grid--center": horizontalAlignment === horizontalAlignmentTypes.center,
		"grid--right": horizontalAlignment === horizontalAlignmentTypes.right,
		"grid--middle": verticalAlignment === verticalAlignmentTypes.middle,
		"grid--bottom": verticalAlignment === verticalAlignmentTypes.bottom,
		"grid--debug": debug,
		[`${className}`]: className
	});

	const GridElementType = elementType || "div";

	return (
		<GridElementType className={classNames} {...rest}>
			{children}
		</GridElementType>
	);
};

// Legacy references - deprecated!
Grid.gutter = gutterTypes;
Grid.horizontalAlignment = horizontalAlignmentTypes;
Grid.verticalAlignment = verticalAlignmentTypes;
