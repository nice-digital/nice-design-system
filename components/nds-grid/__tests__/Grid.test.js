import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import { Grid, GridItem } from "../src/Grid";

describe("Grid", () => {
	it("should render without crashing", () => {
		const wrapper = shallow(
			<Grid>
				<GridItem cols={6}>Test</GridItem>
				<GridItem cols={6}>Test</GridItem>
			</Grid>
		);
		expect(wrapper).toHaveLength(1);
	});

	it("should match snapshot", () => {
		const wrapper = mount(
			<Grid>
				<GridItem cols={6}>Test</GridItem>
				<GridItem cols={6}>Test</GridItem>
			</Grid>
		);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("should add rev class name to grid with reverse prop", () => {
		const wrapper = shallow(
			<Grid reverse>
				<GridItem cols={6}>Test</GridItem>
			</Grid>
		);
		expect(wrapper.find(".grid").props().className).toEqual("grid grid--rev");
	});

	describe("gutters", () => {
		const gutterTest = (gutter, expectedClassName) => {
			const wrapper = shallow(
				<Grid gutter={gutter}>
					<GridItem cols={6}>Test</GridItem>
				</Grid>
			);
			expect(wrapper.find(".grid").props().className).toEqual(
				expectedClassName
			);
		};

		it("shouldn't add any class to grid with gutter prop as standard", () => {
			gutterTest(Grid.gutter.standard, "grid");
		});

		it("should add gutterless class to grid with gutter prop as none", () => {
			gutterTest(Grid.gutter.none, "grid grid--gutterless");
		});

		it("should add compact class to grid with gutter prop as compact", () => {
			gutterTest(Grid.gutter.compact, "grid grid--compact");
		});

		it("should add compact class to grid with gutter prop as compact", () => {
			gutterTest(Grid.gutter.loose, "grid grid--loose");
		});
	});

	describe("horizontal alignment", () => {
		const horizontalAlignmentTest = (hAlignment, expectedClassName) => {
			const wrapper = shallow(
				<Grid horizontalAlignment={hAlignment}>
					<GridItem cols={6}>Test</GridItem>
				</Grid>
			);
			expect(wrapper.find(".grid").props().className).toEqual(
				expectedClassName
			);
		};

		it("shouldn't add any class to grid with horizontalAlignment prop as left", () => {
			horizontalAlignmentTest(Grid.horizontalAlignment.left, "grid");
		});

		it("should add center modifier class to grid when horizontalAlignment prop set to center", () => {
			horizontalAlignmentTest(
				Grid.horizontalAlignment.center,
				"grid grid--center"
			);
		});

		it("should add right modifier class to grid when horizontalAlignment prop set to right", () => {
			horizontalAlignmentTest(
				Grid.horizontalAlignment.right,
				"grid grid--right"
			);
		});
	});

	describe("vertical alignment", () => {
		const verticalAlignmentTest = (vAlignment, expectedClassName) => {
			const wrapper = shallow(
				<Grid verticalAlignment={vAlignment}>
					<GridItem cols={6}>Test</GridItem>
				</Grid>
			);
			expect(wrapper.find(".grid").props().className).toEqual(
				expectedClassName
			);
		};

		it("shouldn't add any class to grid with vertical alignment prop as top", () => {
			verticalAlignmentTest(Grid.verticalAlignment.top, "grid");
		});

		it("should add middle modifier class to grid when verticalAlignment prop set to middle", () => {
			verticalAlignmentTest(Grid.verticalAlignment.middle, "grid grid--middle");
		});

		it("should add bottom modifier class to grid when verticalAlignment prop set to bottom", () => {
			verticalAlignmentTest(Grid.verticalAlignment.bottom, "grid grid--bottom");
		});
	});

	it("should add a debug modifier class when debug prop set to true", () => {
		const wrapper = shallow(
			<Grid debug>
				<GridItem cols={6}>Test</GridItem>
			</Grid>
		);
		expect(wrapper.find(".grid").props().className).toEqual("grid grid--debug");
	});

	it("should add given additional class names to rendered grid class names", () => {
		const wrapper = shallow(
			<Grid className="mt--d">
				<GridItem cols={6}>Test</GridItem>
			</Grid>
		);
		expect(wrapper.find(".grid").props().className).toEqual("grid mt--d");
	});

	it("should use elementType prop as rendered DOM node type", () => {
		const wrapper = shallow(
			<Grid elementType="ul">
				<GridItem cols={6}>Test</GridItem>
			</Grid>
		);
		expect(wrapper.find(".grid").type()).toEqual("ul");
	});
});
