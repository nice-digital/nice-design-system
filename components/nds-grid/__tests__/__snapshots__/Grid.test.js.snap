// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should match snapshot 1`] = `
<Grid
  elementType="div"
  gutter="standard"
  horizontalAlignment="left"
  reverse={false}
  verticalAlignment="top"
>
  <div
    className="grid"
  >
    <GridItem
      cols={6}
      elementType="div"
    >
      <div
        data-g="6"
      >
        Test
      </div>
    </GridItem>
    <GridItem
      cols={6}
      elementType="div"
    >
      <div
        data-g="6"
      >
        Test
      </div>
    </GridItem>
  </div>
</Grid>
`;
